CommonTools/RecoAlgos/interface/FKDPoint.h

```diff
@@ -0,0 +1,98 @@
+#ifndef FKDTREE_KDPOINT_H_
+#define FKDTREE_KDPOINT_H_
+#include <array>
+#include <utility>
+
+template<class TYPE, int numberOfDimensions>
+class FKDPoint
+{
+
+public:
+	FKDPoint() :
+			theElements(), theId(0)
+	{
+	}
+
+	FKDPoint(const FKDPoint<TYPE, numberOfDimensions>& other) :
+			theElements(other.theElements),theId(other.theId)
+	{
+
+	}
+
+	template<class T>
+	FKDPoint<TYPE, numberOfDimensions> & operator=(
+			const FKDPoint<TYPE, numberOfDimensions> & other)
+	{
+		if (this != &other)
+		{
+			theId = other.theId;
+			theElements = other.theElements;
+		}
+		return *this;
+
+	}
```

Comment:
Are the explicit copy constructor and assignment operator needed? I believe these would be the default behaviour anyway.